src/core/types/saved_objects.ts

```diff
@@ -66,7 +68,7 @@ export interface SavedObjectsMigrationVersion {
   [pluginName: string]: string;
 }
 
-export interface SavedObject<T = unknown> {
+export interface SavedObject<T extends SavedObjectAttributes = SavedObjectAttributes> {
```

Comment:
So this is the root change of the PR. We're now longer defaulting to unknown and we're enforcing that `T` extends a serializable object/record.